### DISCUSSION TITLE
Python Version Poll - please vote!

### DISCUSSION THREAD

**Discussion by eavanvalkenburg** [microsoft/semantic-kernel#5691] [Polls]
Hi all,

We are debating which python versions we should support, currently we run tests on 3.8 - 3.12. But 3.8 is going out of support in October, and 3.9 a year later. But for the kernelfunction from method we rely on the types of the parameters to properly translate those in, add the annotations (for function calling) and do model validation if you want to use a custom class as a parameter and still have function calling give you the right input, so we want to know which version you are using most of the time with Semantic Kernel!

Let us know!

**Comment by timrobertsusa** [comment]
from within vscode-- but I vote more the 3.12

**Comment by lovedeepatsgit** [comment]
Earlier I am using 3.11, but I to use Prompt Flow and Semantic Kernel together on Azure AI Foundry I have to switch to 3.10.1.

**Reply by eavanvalkenburg** [reply]
@lovedeepatsgit currently Semantic Kernel runs on 3.10 - 3.12 (and probable on 3.13 as well but not all dependencies are capable of that yet, so will depend on what you need)

**Reply by lovedeepatsgit** [reply]
Yeah @eavanvalkenburg SK is working nicely with Kernel Functions on python 3.10.1. But you are absolutely right it depends on what we need.
